components/constellation/constellation.rs

```diff
@@ -992,6 +999,40 @@ where
             FromCompositorMsg::Keyboard(key_event) => {
                 self.handle_key_msg(key_event);
             },
+            // Perform a navigation previously requested by script, if approved by the embedder.
+            // If there is already a pending page (self.pending_changes), it will not be overridden;
+            // However, if the id is not encompassed by another change, it will be.
+            FromCompositorMsg::AllowNavigation(
+                top_level_browsing_context_id,
+                pipeline_id,
+                url,
+                allowed,
+            ) => {
+                if !allowed {
+                    return;
```

Comment:
Don't you need to remove the pending navigation from pending_approval_navigations?